### ISSUE TITLE
allow multiple instances of Robomongo [Feature request]

### ISSUE THREAD

**Issue by marmor7** [paralect/robomongo#1247]
Sometimes I have multiple shells open on our staging environment, and multiple shells on production, with similar named collections in both, and it gets confusing.

I would be nice to be allowed to open two or more instances of Robomongo, where I could open different types of shells on each.

While I'm at it, being able to save and open "workspaces"/"projects" of multiple shells with their contents would be nice to have too (e.g. like projects in Sublime).

**Comment by juliashibalko** [comment]
Hi @marmor7, thanks a lot for interest in Robomongo! This feature really makes sense and is already in our to-do list. If I'm not mistaken, here is same ticket with issue description #1161. So let me suggest you to check progress within this one. I'm closing ticket as duplicate (but feel free to reopen it in any case). We will get back to you with any updates, stay tuned!

**Comment by marmor7** [comment]
Hey Julia, I don't think this is the same feature.
The feature you've mentioned is referring to running multiple queries at the same time, I'm referring to being able to open multiple windows of Robomongo at the same time.

Like many apps on mac/windows, you can open multiple instances of the app, for example in chrome you can have a window logged in with your private account, and another window with your company profile, each having their own set of open tabs.

In Robomongo I can't seem to open more then one instance, so my staging and production shells are mixed up together.

**Comment by juliashibalko** [comment]
@marmor7 Ah, I see your concern. Maybe the ticket title is not exact. May I ask you to check this page about [Multi-process architecture](http://blog.robomongo.org/mongodb-shell-integration/)? Seems that here is solution that you're talking about - to have separate shell process (like have  many mongo shell opened separately). Thanks in advance for the answer!

**Comment by simsekgokhan** [comment]
@marmor7 , I agree something like chrome, private and non-private design can be very useful.
I think the feature in Julia's last comment will help achieving your need.

About this "I would be nice to be allowed to open two or more instances of Robomongo", I am assuming you use MAC or Linux since it is very easy to open multiple instance of same program on Windows by simply clicking it. And by design of OSX, it is not very easy.

For now, on MAC/Linux, would these help?

```
How to open multiple instance of a program?
// Mac
open -n /Applications/Robomongo.app
// Linux (ubuntu)
Click the icon of the running application with the middle mouse button.
```

**Comment by marmor7** [comment]
Thanks @simsekgokhan i'm indeed using mac, and `open -n` did the trick, and suits my request.

However, it's behaving a bit weird, and unlike other mac apps.
Usually starting a new window for the current app can be done in a menu option, or command+N, then you can toggle different apps via command+tab, and toggle different windows of the same app via command+`.
When opening via `open -n` it behaves like two different apps, toggling between windows of Robomongo when using command+tab, as well as displaying two icons of Robomongo in the bottom dock.

**Comment by simsekgokhan** [comment]
I think those are expected behaviors since currently there is no special design of Robomongo for this case. And the fact that there is no easy way to open two programs (we do it through command line is kind of hacky way), shows that this use case isn't something OSX directly supports or encourages.

Do you have any major problems with Robomongo when you open multiple instances on command line?

**Comment by lesterjanpearson** [comment]
Is there a keyboard shortcut for this?

**Comment by simsekgokhan** [comment]
Hi @lesterjanpearson , since this behavior is not officially supported by OS X design, I do not think so.

Another solution might be creating a very simple run-able script as explained below.
Place the file "robo.command" into some place in your desktop and every double click should open a new Robomongo instance. (Note that "command" file extension is required.)

```
cd ~/Desktop
vim or gedit robo.command
chmod +x robo.command 
```

**'robo.command' file:** 
```
#!/usr/bin/env bash
open -n /Applications/Robomongo.app/
```

**Comment by danielflippance** [comment]
+1

**Comment by afeldman1** [comment]
The label "works as designed" must be incorrect. If you try to connect to multiple mongodb databases that use the same replica name, robo 3T provides the below error message. The message(relevant part bolded) clearly states that multiple instances are not only supported but required for this functionality. We should either change the error message to state that this real use case is not currently supported and re-open this bug based on the error message or create a new issue for allowing a single robomongo instance to connect to multiple mongodb servers with the same replica name.

```
Cannot connect to replica set ********. 

A primary with different host name [*******] found in server side. Please double check if same host names and ports are used as in server's replica set configuration. 
If same set name is used for different replica sets, this configuration is supported only on different instances of Robomongo.  **Please open a new Robomongo instance** for each replica set which has the same set name.

Reason:
Different members found under same replica set name "********.".
```
![image](https://user-images.githubusercontent.com/4326577/85193155-ffd04f80-b293-11ea-8c3e-71ab748e6aae.png)

**Comment by afeldman1** [comment]
Related: #1513 .This issue seems to be reported many times throughout the years
